apps/jobs/views.py

```diff
@@ -372,6 +372,19 @@ def change_submission_data_and_visibility(
         if is_public is True:
             when_made_public = datetime.datetime.now()
             request.data["when_made_public"] = when_made_public
+
+        submissions_already_public = Submission.objects.filter(
+            is_public=True,
+            participant_team=participant_team,
+            challenge_phase=challenge_phase
+        )
+
+        if (challenge_phase.is_restricted_to_select_one_submission
```

Comment:
Can we please add a comment before this line so that we remember later why this check was added ?